snap/local/runtime-helpers/bin/install-setup-postgres.sh

```diff
@@ -1,73 +1,114 @@
 #!/bin/sh
 
+# this script is called from the install hook to set up postgres
+# and also from the post-refresh hook. It handles these three scenerios:
+# - new install: create config file and database
+# - refresh from 2.0/stable or older - a snap using postgres 10
+# - refresh from a 2.1/stable or newer snap using postgres 12
+
+# in postgres 12 we are using these directories
+SNAP_DATA_CONFIG_POSTGRES_DIR="$SNAP_DATA/config/postgres"
+SNAP_DATA_CONFIG_POSTGRES_CONFIGFILE="$SNAP_DATA_CONFIG_POSTGRES_DIR/postgresql.conf"
+SNAP_DATA_POSTGRES_DIR="$SNAP_DATA/postgresql"
+SNAP_DATA_POSTGRES_DB_DIR="$SNAP_DATA_POSTGRES_DIR/data"
```

Comment:
I wonder if we should include the postgres version number in the directory path here, if we need to start supporting more versions of postgres when i.e. we move to `base: core22` in X years

Reply:
Updated this to include the version numbers